components/sidebar/channel_navigator/channel_navigator.tsx

```diff
@@ -78,21 +79,25 @@ export default class ChannelNavigator extends React.PureComponent<Props> {
             </button>
         );
 
-        const addChannelDropdown = (
-            <AddChannelDropdown
-                showNewChannelModal={this.props.showNewChannelModal}
-                showMoreChannelsModal={this.props.showMoreChannelsModal}
-                invitePeopleModal={this.props.invitePeopleModal}
-                showCreateCategoryModal={this.props.showCreateCategoryModal}
-                canCreateChannel={this.props.canCreateChannel}
-                canJoinPublicChannel={this.props.canJoinPublicChannel}
-                handleOpenDirectMessagesModal={this.props.handleOpenDirectMessagesModal}
-                unreadFilterEnabled={this.props.unreadFilterEnabled}
-                townSquareDisplayName={this.props.townSquareDisplayName}
-                offTopicDisplayName={this.props.offTopicDisplayName}
-                showTutorialTip={this.props.showTutorialTip}
-            />
-        );
+        let addChannelDropdown = null;
+        if (!this.props.addChannelButton || this.props.addChannelButton === AddChannelButtonTreatments.NONE) {
```

Comment:
@neallred Here we should check if the Global Header is off? Currently, the `AddChannelButtonTreatments` is set to the default we configured for 6.0 https://github.com/mattermost/mattermost-server/pull/18281 and if `GlobalHeader` is off then it seems to be causing https://mattermost.atlassian.net/browse/MM-38612

Reply:
I agree; we should check and I think it is the cause of the bug. I can take the bug up.